File: src/main/java/ch/zxseitz/j3de/windows/KeyActionType.java
package ch.zxseitz.j3de.windows;

import static org.lwjgl.glfw.GLFW.*;

public enum KeyActionType {
    UNKNOWN,
    UP,
    DOWN,
    PRESSED;

    public static KeyActionType fromValue(int value) {
        return switch (value) {
            case GLFW_RELEASE -> UP;
            case GLFW_PRESS -> DOWN;
            case GLFW_REPEAT -> PRESSED;
            default -> UNKNOWN;
        };
    }
}

File: src/main/java/ch/zxseitz/j3de/window/KeyActionType.kt
package ch.zxseitz.j3de.window;

import org.lwjgl.glfw.GLFW.*;

enum class KeyActionType {
    UP,
    DOWN,
    PRESSED;

    companion object {
        fun fromValue(value: Int): KeyActionType? {
            return when(value) {
                GLFW_RELEASE -> UP;
                GLFW_PRESS -> DOWN;
                GLFW_REPEAT -> PRESSED;
                else -> null;
            }
        }
    }
}
